src/puppetlabs/puppetdb/query/facts.clj

```diff
@@ -31,26 +28,22 @@
 
 ;; MUNGE
 
-(pls/defn-validated convert-types :- [converted-row-schema]
+(pls/defn-validated deserialize-fact-value :- [converted-row-schema]
   "Coerce values for each row to the proper stored type."
   [rows :- [row-schema]]
-  (map (partial facts/convert-row-type [:type :depth :value_integer :value_float]) rows))
+  (map #(utils/update-when % [:value] json/parse-string) rows))
 
-(pls/defn-validated munge-result-rows
-  [_
-   projected-fields :- [s/Keyword]
-   _
-   _]
+(defn munge-result-rows
+  [_ _ _]
   (fn [rows]
     (if (empty? rows)
       []
       (->> rows
-        convert-types
-        (map #(select-keys % (or (seq projected-fields)
-                                 [:certname :environment :timestamp :name :value])))))))
+           deserialize-fact-value
+           (map identity)))))
```

Comment:
@senior this is the line in question. I don't know why removing it causes the services test to fail. Validating on the whole collection instead of element by element in the deserialization function seems like a potential issue too -- I want to touch that up before this gets merged.